osf_tests/test_file_metadata.py

```diff
@@ -0,0 +1,189 @@
+import json
+import pytest
+import jsonschema
+from django.contrib.contenttypes.models import ContentType
+
+from framework.auth.core import Auth
+from framework.exceptions import PermissionsError
+from website.project.licenses import set_license
+from osf.models import FileMetadataSchema, NodeLicense, Guid, NodeLog
+from osf_tests.factories import ProjectFactory, SubjectFactory, AuthUserFactory
+from api_tests.utils import create_test_file
+
+
+@pytest.fixture()
+def node():
+    return ProjectFactory()
+
+@pytest.fixture()
+def osf_file(node):
+    return create_test_file(target=node, user=node.creator)
+
+@pytest.mark.django_db
+class TestFileMetadataRecordSerializer:
+
+    def test_record_created_post_save(self, node, osf_file):
+        # check there's a record for every FileMetadataSchema
+        assert FileMetadataSchema.objects.count() > 0
+        assert osf_file.records.count() == FileMetadataSchema.objects.count()
+
+        for record in osf_file.records.all().select_related('file'):
+            assert record.file == osf_file
+
+    def test_serialize_record_datacite(self, node, osf_file):
+        # Test all of the parts of serialize_json that are auto-generated
+        # from relationships and properties on the node and file
+
+        # add a contributor with an ORCID
+        contributor = AuthUserFactory()
+        contributor.external_identity = {
+            'ORCID': {
+                '0000-0001-9143-4653': 'VERIFIED'
+            }
+        }
+        contributor.save()
+        node.add_contributor(contributor, save=False)
+
+        # add subjects, tags, license, and guid
+        tags = ['fish', 'scale']
+        [osf_file.add_tag(tag, auth=Auth(node.creator), save=False) for tag in tags]
+        bepress_subject = SubjectFactory(text='BePress Text')
+        new_subject = SubjectFactory(bepress_subject=bepress_subject)
+        node.subjects.add(new_subject)
+        no_license = NodeLicense.objects.get(name='CC0 1.0 Universal')
+        license_detail = {
+            'id': no_license.license_id,
+            'year': '2018',
+            'copyrightHolders': ['Woop', 'Yeah']
+        }
+        set_license(node, license_detail, Auth(node.creator))
+        osf_file.save()
+        node.save()
+
+        # create a guid
```

Comment:
@erinspace, I don't think I follow -- why is this necessary?

Reply:
...I have no recollection of why I added that, and nothing breaks if I remove it. Weird!